compiler/src/typed/cmi_format.re

```diff
@@ -142,22 +138,32 @@ module CmiBinarySection =
     let accepts_version = ({major}) => major == 1;
   });
 
-let read_cmi = filename => {
-  let ic = open_in_bin(filename);
-  switch (CmiBinarySection.load(ic)) {
-  | Some(cmi) =>
-    close_in(ic);
-    cmi;
-  | None
-  | exception End_of_file
-  | exception (Invalid_argument(_))
-  | exception (Failure(_)) =>
-    close_in(ic);
+let read_cmi = (ic, filename): cmi_infos => {
+  let read_magic = really_input_string(ic, 4);
+  if (read_magic != magic) {
     raise(Error(Corrupted_interface(filename)));
-  | exception (Error(e)) =>
-    close_in(ic);
-    raise(Error(e));
   };
+  let version_length = input_binary_int(ic);
+  let read_version = really_input_string(ic, version_length);
+  if (read_version != Config.version) {
+    raise(Error(Wrong_version_interface(filename, read_version)));
+  };
+  let _ = input_binary_int(ic);
```

Comment:
Why not do the same validation as lines 48-51 of emitmod where you verify the length is correct?

Reply:
If it's wrong, `Marshal.from_channel` will raise, which we catch and raise `Corrupted_interface`. We could check it, but it'd be redundant and also make us read the entire channel.